aya-obj/src/obj.rs

```diff
@@ -1280,8 +1287,26 @@ fn parse_btf_map_def(btf: &Btf, info: &DataSecEntry) -> Result<(String, BtfMapDe
         }
     };
 
+    let mut map_def = BtfMapDef::default();
+
     for m in &s.members {
+        // In aya-ebpf, the BTF map definition types are not used directly.
+        // Instead, they are wrapped in structs provided by aya-ebpf (e.g.
+        // `HashMap`, `Array`), which then wrap the definition type in
+        // `UnsafeCell`.
+        // To retrieve the definition type, we need to walk through all the
+        // wrapper types:
+        //
+        // - aya-ebpf wrappers like `HashMap`, `Array` etc. They wrap an
+        //   `UnsafeCell` in a tuple struct with one member, hence the field
+        //   name is `__0`.
+        // - `UnsafeCell`, which wraps the BTF map definition inside a `value`
+        //   field.
         match btf.string_at(m.name_offset)?.as_ref() {
+            "__0" => {
```

Comment:
Because of this quirk, the eBPF programs written with aya-ebpf are **not** loadable by libbpf. We would have to port exactly the same quirk there.

But I didn't see any issues on the kernel side with it.